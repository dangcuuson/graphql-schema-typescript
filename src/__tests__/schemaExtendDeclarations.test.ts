import * as path from 'path';
import { generateTypeScriptTypes } from '..';
import { executeApiTest } from './testUtils';

describe('schema extension declaration', () => {
    
    it('should generate if schema is a valid graphql schema', async () => {
        const schemaStr = `schema { query: Query } type Query { test: String! } extend type Query { foo: String }`;
        const generated = await executeApiTest('schemaString.ts', {}, schemaStr);
        expect(generated).toEqual(expect.stringContaining('foo?: string'));
    });
});