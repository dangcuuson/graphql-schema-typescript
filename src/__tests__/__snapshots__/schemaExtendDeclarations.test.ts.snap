// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema extension declaration should generate if schema is a valid graphql schema 1`] = `
"/* tslint:disable */
/* eslint-disable */
import { GraphQLResolveInfo } from 'graphql';
/**
 * This file is auto-generated by @raraujo/graphql-schema-typescript
 * Please note that any changes in this file may be overwritten
 */
 

/*******************************
 *                             *
 *          TYPE DEFS          *
 *                             *
 *******************************/
export interface GQLQuery {
  test: string;
  foo?: string;
}

/*********************************
 *                               *
 *         TYPE RESOLVERS        *
 *                               *
 *********************************/
/**
 * This interface define the shape of your resolver
 * Note that this type is designed to be compatible with graphql-tools resolvers
 * However, you can still use other generated interfaces to make your resolver type-safed
 */
export interface GQLResolver {
  Query?: GQLQueryTypeResolver;
}
export interface GQLQueryTypeResolver<TParent = any> {
  test?: QueryToTestResolver<TParent>;
  foo?: QueryToFooResolver<TParent>;
}

export interface QueryToTestResolver<TParent = any, TResult = any> {
  (parent: TParent, args: {}, context: any, info: GraphQLResolveInfo): TResult;
}

export interface QueryToFooResolver<TParent = any, TResult = any> {
  (parent: TParent, args: {}, context: any, info: GraphQLResolveInfo): TResult;
}
"
`;
